libcoap/src/message/request.rs

```diff
@@ -428,7 +428,7 @@ impl CoapRequest {
     /// Converts this request into a [CoapMessage] that can be sent over a [CoapSession](crate::session::CoapSession).
     pub fn into_message(mut self) -> CoapMessage {
         if self.uri.is_proxy() {
-            self.pdu.add_option(CoapOption::ProxyUri(
+            self.pdu.add_option(CoapOption::ProxyScheme(
```

Comment:
Hmm, is this change related to the main fix of this PR? Otherwise, it could also become its own commit, I guess.